ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckOperation.java

```diff
@@ -75,9 +76,9 @@ public int execute() throws HiveException, IOException, TException {
           desc.getFilterExp(), desc.getResFile(), desc.isRepairPartitions(),
           desc.isAddPartitions(), desc.isDropPartitions(), partitionExpirySeconds);
       return msck.repair(msckInfo);
-    } catch (MetaException e) {
+    } catch (MetaException | MetastoreException e) {
       LOG.error("Unable to create msck instance.", e);
-      return 1;
+      throw e;
     } catch (SemanticException e) {
```

Comment:
Will client get the detail error information in case of `SemanticException`?

Reply:
Yes. Added test case msck_repair_8.q to see how semantic exception is handled